src/ucp/wireup/select.c

```diff
@@ -1377,20 +1423,26 @@ ucp_wireup_construct_lanes(ucp_wireup_select_ctx_t *select_ctx,
     key->am_bw_lanes[0] = key->am_lane;
 }
 
-ucs_status_t ucp_wireup_select_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
-                                     unsigned ep_init_flags, unsigned address_count,
+ucs_status_t ucp_wireup_select_lanes(ucp_ep_h ep,
+                                     const ucp_ep_params_t *ep_params,
+                                     unsigned ep_init_flags,
+                                     unsigned address_count,
                                      const ucp_address_entry_t *address_list,
-                                     uint8_t *addr_indices, ucp_ep_config_key_t *key)
+                                     uint8_t *addr_indices,
+                                     ucp_ep_config_key_t *key)
 {
     ucp_worker_h worker = ep->worker;
-    ucp_wireup_select_ctx_t select_ctx;
+    ucp_wireup_select_context_t select_ctx;
+    ucp_wireup_select_params_t select_params;
     ucs_status_t status;
 
     if (worker->scalable_tl_bitmap) {
-        ucp_wireup_select_ctx_init(&select_ctx, ep, params, ep_init_flags,
-                                   address_count, address_list,
-                                   worker->scalable_tl_bitmap);
-        status = ucp_wireup_search_lanes(&select_ctx, key);
+        ucp_wireup_select_params_init(&select_params, ep, ep_params,
+                                      ep_init_flags, address_count,
+                                      address_list, worker->scalable_tl_bitmap);
+        memset(&select_ctx, 0, sizeof(select_ctx));
```

Comment:
move `memset(&select_ctx, 0, sizeof(select_ctx));` to `ucp_wireup_search_lanes()`?

Reply:
ok, I was hesitate about this

Reply:
> ok, I was hesitate about this

it's up to you anyway